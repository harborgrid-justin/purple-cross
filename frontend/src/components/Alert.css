/* Alert Component Styles */

.alert {
  padding: var(--spacing-md, 1rem) var(--spacing-lg, 1.5rem);
  border-radius: var(--radius-md, 0.5rem);
  border-left: 4px solid;
  display: flex;
  align-items: flex-start;
  gap: var(--spacing-md, 1rem);
  margin-bottom: var(--spacing-md, 1rem);
}

.alert-icon {
  flex-shrink: 0;
  font-size: 1.25rem;
  line-height: 1;
}

.alert-content {
  flex: 1;
  min-width: 0;
}

.alert-title {
  margin: 0 0 0.25rem;
  font-weight: 600;
  font-size: 0.9375rem;
  line-height: 1.5;
}

.alert-message {
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.5;
}

.alert-close {
  flex-shrink: 0;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0.25rem;
  border-radius: var(--radius-sm, 0.25rem);
  color: inherit;
  opacity: 0.6;
  transition: opacity 0.2s;
  font-size: 1.25rem;
  line-height: 1;
  margin: -0.25rem -0.5rem -0.25rem 0;
}

.alert-close:hover {
  opacity: 1;
}

.alert-close:focus-visible {
  outline: 2px solid currentColor;
  outline-offset: 2px;
  opacity: 1;
}

/* Alert Variants */
.alert-success {
  background-color: #d1fae5;
  border-color: #10b981;
  color: #065f46;
}

.alert-error {
  background-color: #fee2e2;
  border-color: #ef4444;
  color: #991b1b;
}

.alert-warning {
  background-color: #fef3c7;
  border-color: #f59e0b;
  color: #92400e;
}

.alert-info {
  background-color: #dbeafe;
  border-color: #3b82f6;
  color: #1e40af;
}

/* Dismissible Alert */
.alert-dismissible {
  padding-right: var(--spacing-xl, 2rem);
}

/* Compact Variant */
.alert-compact {
  padding: var(--spacing-sm, 0.5rem) var(--spacing-md, 1rem);
}

.alert-compact .alert-icon {
  font-size: 1rem;
}

.alert-compact .alert-title {
  font-size: 0.875rem;
}

.alert-compact .alert-message {
  font-size: 0.8125rem;
}

/* Responsive */
@media (max-width: 768px) {
  .alert {
    padding: var(--spacing-sm, 0.5rem) var(--spacing-md, 1rem);
    gap: var(--spacing-sm, 0.5rem);
  }

  .alert-icon {
    font-size: 1rem;
  }
}

/* Animation for entering/leaving */
@keyframes alertSlideIn {
  from {
    opacity: 0;
    transform: translateY(-10px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
}

.alert {
  animation: alertSlideIn 0.2s ease-out;
}

@media (prefers-reduced-motion: reduce) {
  .alert {
    animation: none;
  }
}
