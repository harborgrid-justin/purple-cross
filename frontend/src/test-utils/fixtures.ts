/**
 * WF-COMP-XXX | fixtures.ts - Test Fixtures
 * Purpose: Mock data fixtures for testing
 * Dependencies: None
 * Last Updated: 2025-10-24 | File Type: .ts
 */

/**
 * Mock patient data
 */
export const mockPatient = {
  id: '1',
  name: 'Buddy',
  species: 'Dog',
  breed: 'Golden Retriever',
  dateOfBirth: '2020-01-15',
  gender: 'Male',
  color: 'Golden',
  weight: 32.5,
  microchipNumber: '123456789012345',
  status: 'active',
  clientId: '1',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

export const mockPatients = [
  mockPatient,
  {
    id: '2',
    name: 'Whiskers',
    species: 'Cat',
    breed: 'Siamese',
    dateOfBirth: '2019-05-20',
    gender: 'Female',
    color: 'Cream',
    weight: 4.2,
    microchipNumber: '987654321098765',
    status: 'active',
    clientId: '1',
    createdAt: '2025-01-01T00:00:00Z',
    updatedAt: '2025-01-01T00:00:00Z',
  },
  {
    id: '3',
    name: 'Max',
    species: 'Dog',
    breed: 'German Shepherd',
    dateOfBirth: '2018-03-10',
    gender: 'Male',
    color: 'Black and Tan',
    weight: 38.0,
    status: 'inactive',
    clientId: '2',
    createdAt: '2025-01-01T00:00:00Z',
    updatedAt: '2025-01-01T00:00:00Z',
  },
];

/**
 * Mock client data
 */
export const mockClient = {
  id: '1',
  firstName: 'John',
  lastName: 'Doe',
  email: 'john.doe@example.com',
  phone: '555-0100',
  address: '123 Main St',
  city: 'Springfield',
  state: 'IL',
  zipCode: '62701',
  status: 'active',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

export const mockClients = [
  mockClient,
  {
    id: '2',
    firstName: 'Jane',
    lastName: 'Smith',
    email: 'jane.smith@example.com',
    phone: '555-0200',
    address: '456 Oak Ave',
    city: 'Springfield',
    state: 'IL',
    zipCode: '62702',
    status: 'active',
    createdAt: '2025-01-01T00:00:00Z',
    updatedAt: '2025-01-01T00:00:00Z',
  },
];

/**
 * Mock appointment data
 */
export const mockAppointment = {
  id: '1',
  patientId: '1',
  clientId: '1',
  staffId: '1',
  appointmentType: 'checkup',
  startTime: '2025-10-25T10:00:00Z',
  endTime: '2025-10-25T10:30:00Z',
  status: 'scheduled',
  notes: 'Annual wellness exam',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

export const mockAppointments = [
  mockAppointment,
  {
    id: '2',
    patientId: '2',
    clientId: '1',
    staffId: '1',
    appointmentType: 'surgery',
    startTime: '2025-10-25T14:00:00Z',
    endTime: '2025-10-25T15:00:00Z',
    status: 'confirmed',
    notes: 'Spay procedure',
    createdAt: '2025-01-01T00:00:00Z',
    updatedAt: '2025-01-01T00:00:00Z',
  },
];

/**
 * Mock medical record data
 */
export const mockMedicalRecord = {
  id: '1',
  patientId: '1',
  staffId: '1',
  visitDate: '2025-10-20',
  chiefComplaint: 'Annual checkup',
  diagnosis: 'Healthy',
  treatment: 'Vaccinations administered',
  prescriptions: [],
  labTests: [],
  status: 'completed',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

export const mockMedicalRecords = [
  mockMedicalRecord,
  {
    id: '2',
    patientId: '1',
    staffId: '1',
    visitDate: '2025-09-15',
    chiefComplaint: 'Ear infection',
    diagnosis: 'Otitis externa',
    treatment: 'Antibiotic ear drops',
    prescriptions: ['1'],
    labTests: [],
    status: 'completed',
    createdAt: '2025-01-01T00:00:00Z',
    updatedAt: '2025-01-01T00:00:00Z',
  },
];

/**
 * Mock prescription data
 */
export const mockPrescription = {
  id: '1',
  patientId: '1',
  medicationName: 'Amoxicillin',
  dosage: '250mg',
  frequency: 'Twice daily',
  duration: '10 days',
  instructions: 'Give with food',
  prescribedBy: '1',
  status: 'active',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

/**
 * Mock staff data
 */
export const mockStaff = {
  id: '1',
  firstName: 'Dr. Sarah',
  lastName: 'Johnson',
  email: 'sarah.johnson@purplecross.com',
  phone: '555-0300',
  role: 'veterinarian',
  specialty: 'General Practice',
  licenseNumber: 'VET-12345',
  status: 'active',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

/**
 * Mock invoice data
 */
export const mockInvoice = {
  id: '1',
  clientId: '1',
  patientId: '1',
  invoiceNumber: 'INV-2025-001',
  invoiceDate: '2025-10-20',
  dueDate: '2025-11-20',
  subtotal: 150.0,
  tax: 12.0,
  total: 162.0,
  amountPaid: 0.0,
  status: 'unpaid',
  items: [
    {
      description: 'Annual wellness exam',
      quantity: 1,
      unitPrice: 100.0,
      total: 100.0,
    },
    {
      description: 'Rabies vaccination',
      quantity: 1,
      unitPrice: 50.0,
      total: 50.0,
    },
  ],
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

/**
 * Mock inventory item data
 */
export const mockInventoryItem = {
  id: '1',
  name: 'Amoxicillin 250mg',
  category: 'Medication',
  sku: 'MED-001',
  quantity: 100,
  reorderLevel: 20,
  unitPrice: 2.5,
  supplier: 'VetSupply Co',
  expirationDate: '2026-12-31',
  status: 'active',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

/**
 * Mock lab test data
 */
export const mockLabTest = {
  id: '1',
  patientId: '1',
  testType: 'Blood panel',
  testDate: '2025-10-20',
  results: 'All values within normal range',
  status: 'completed',
  priority: 'routine',
  orderedBy: '1',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

/**
 * Mock document data
 */
export const mockDocument = {
  id: '1',
  patientId: '1',
  clientId: '1',
  title: 'Vaccination Certificate',
  documentType: 'Certificate',
  fileName: 'vaccination-cert.pdf',
  fileSize: 245678,
  mimeType: 'application/pdf',
  uploadedBy: '1',
  status: 'signed',
  createdAt: '2025-01-01T00:00:00Z',
  updatedAt: '2025-01-01T00:00:00Z',
};

/**
 * Mock user data
 */
export const mockUser = {
  id: '1',
  email: 'test@example.com',
  firstName: 'Test',
  lastName: 'User',
  role: 'veterinarian',
  token: 'mock-jwt-token',
};

/**
 * Mock API response
 */
export const mockApiResponse = <T>(data: T) => ({
  status: 'success',
  data,
});

/**
 * Mock API error response
 */
export const mockApiError = (message: string, statusCode = 400) => ({
  status: 'error',
  message,
  statusCode,
});

/**
 * Mock pagination metadata
 */
export const mockPagination = {
  page: 1,
  limit: 10,
  total: 100,
  totalPages: 10,
};

/**
 * Generate mock list with pagination
 */
export const mockPaginatedResponse = <T>(
  data: T[],
  page = 1,
  limit = 10,
  total?: number
) => ({
  status: 'success',
  data,
  pagination: {
    page,
    limit,
    total: total || data.length,
    totalPages: Math.ceil((total || data.length) / limit),
  },
});
