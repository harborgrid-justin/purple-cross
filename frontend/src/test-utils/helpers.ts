/**
 * WF-COMP-XXX | helpers.ts - Test Helper Functions
 * Purpose: Common utilities for testing
 * Dependencies: @testing-library/react
 * Last Updated: 2025-10-24 | File Type: .ts
 */

import { waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

/**
 * Create a user event instance for simulating user interactions
 * This should be called at the beginning of each test
 *
 * @example
 * ```tsx
 * const user = createUser();
 * await user.click(button);
 * await user.type(input, 'text');
 * ```
 */
export const createUser = () => {
  return userEvent.setup();
};

/**
 * Wait for an async operation to complete with custom timeout
 */
export const waitForAsync = async (
  callback: () => void | Promise<void>,
  timeout = 3000
): Promise<void> => {
  await waitFor(callback, { timeout });
};

/**
 * Wait for loading states to complete
 * Useful for components that show loading spinners
 */
export const waitForLoadingToFinish = async (
  getByText: (text: string | RegExp) => HTMLElement,
  timeout = 3000
): Promise<void> => {
  await waitFor(
    () => {
      expect(getByText(/loading/i)).not.toBeInTheDocument();
    },
    { timeout }
  );
};

/**
 * Mock window.scrollTo (not implemented in jsdom)
 */
export const mockScrollTo = (): void => {
  window.scrollTo = vi.fn();
};

/**
 * Mock localStorage for testing
 */
export const mockLocalStorage = (): {
  getItem: ReturnType<typeof vi.fn>;
  setItem: ReturnType<typeof vi.fn>;
  removeItem: ReturnType<typeof vi.fn>;
  clear: ReturnType<typeof vi.fn>;
} => {
  const storage: Record<string, string> = {};

  return {
    getItem: vi.fn((key: string) => storage[key] || null),
    setItem: vi.fn((key: string, value: string) => {
      storage[key] = value;
    }),
    removeItem: vi.fn((key: string) => {
      delete storage[key];
    }),
    clear: vi.fn(() => {
      Object.keys(storage).forEach((key) => delete storage[key]);
    }),
  };
};

/**
 * Mock sessionStorage for testing
 */
export const mockSessionStorage = mockLocalStorage;

/**
 * Delay execution for a specified time (in milliseconds)
 * Useful for testing loading states or debounced functions
 */
export const delay = (ms: number): Promise<void> => {
  return new Promise((resolve) => setTimeout(resolve, ms));
};

/**
 * Create a mock file for file upload testing
 */
export const createMockFile = (
  name: string,
  content: string,
  type: string
): File => {
  const blob = new Blob([content], { type });
  return new File([blob], name, { type });
};

/**
 * Create a mock image file for image upload testing
 */
export const createMockImageFile = (
  name = 'test-image.png',
  size = 1024
): File => {
  const buffer = new ArrayBuffer(size);
  const blob = new Blob([buffer], { type: 'image/png' });
  return new File([blob], name, { type: 'image/png' });
};

/**
 * Assert that an element has a specific class
 */
export const expectToHaveClass = (
  element: HTMLElement,
  className: string
): void => {
  expect(element.classList.contains(className)).toBe(true);
};

/**
 * Assert that an element does not have a specific class
 */
export const expectNotToHaveClass = (
  element: HTMLElement,
  className: string
): void => {
  expect(element.classList.contains(className)).toBe(false);
};

/**
 * Get computed style of an element
 */
export const getComputedStyleValue = (
  element: HTMLElement,
  property: string
): string => {
  return window.getComputedStyle(element).getPropertyValue(property);
};

/**
 * Simulate a network delay for testing loading states
 */
export const simulateNetworkDelay = async (ms = 500): Promise<void> => {
  await delay(ms);
};

/**
 * Create a spy for console methods (useful for testing error handling)
 */
export const spyOnConsole = (): {
  log: ReturnType<typeof vi.spyOn>;
  warn: ReturnType<typeof vi.spyOn>;
  error: ReturnType<typeof vi.spyOn>;
} => {
  return {
    log: vi.spyOn(console, 'log').mockImplementation(() => {}),
    warn: vi.spyOn(console, 'warn').mockImplementation(() => {}),
    error: vi.spyOn(console, 'error').mockImplementation(() => {}),
  };
};

/**
 * Assert that a function was called with specific arguments
 */
export const expectCalledWith = <T extends (...args: any[]) => any>(
  fn: T,
  ...args: Parameters<T>
): void => {
  expect(fn).toHaveBeenCalledWith(...args);
};

/**
 * Wait for a specific number of milliseconds in tests
 * Uses vi.advanceTimersByTime if fake timers are enabled
 */
export const advanceTime = async (ms: number): Promise<void> => {
  if (vi.isFakeTimers()) {
    vi.advanceTimersByTime(ms);
  } else {
    await delay(ms);
  }
};

/**
 * Type assertion helper for better type inference in tests
 */
export const assertType = <T>(_value: T): void => {
  // Type assertion helper - no runtime behavior
};
