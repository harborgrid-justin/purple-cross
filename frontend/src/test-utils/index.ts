/**
 * WF-COMP-XXX | index.ts - Test Utils Exports
 * Purpose: Central export point for all test utilities
 * Dependencies: None
 * Last Updated: 2025-10-24 | File Type: .ts
 */

// Export custom render functions
export * from './render';

// Export test helpers
export * from './helpers';

// Export test fixtures
export * from './fixtures';

// Export MSW mocks
export * from './mocks/handlers';
export { server, mockServer } from './mocks/server';
