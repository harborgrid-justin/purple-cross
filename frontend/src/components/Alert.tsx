/**
 * WF-COMP-XXX | Alert.tsx - Alert
 * Purpose: React component for Alert functionality
 * Dependencies: react
 * Last Updated: 2025-10-23 | File Type: .tsx
 */

import { ReactNode } from 'react';
import clsx from 'clsx';
import './Alert.css';

interface AlertProps {
  type?: 'success' | 'error' | 'warning' | 'info';
  children: ReactNode;
  onClose?: () => void;
  className?: string;
}

const Alert = ({ type = 'info', children, onClose, className }: AlertProps) => {
  const icons = {
    success: '✓',
    error: '✕',
    warning: '⚠',
    info: 'ℹ',
  };

  return (
    <div className={clsx('alert', `alert-${type}`, className)} role="alert">
      <span className="alert-icon" aria-hidden="true">
        {icons[type]}
      </span>
      <div className="alert-content">{children}</div>
      {onClose && (
        <button className="alert-close" onClick={onClose} aria-label="Close alert" type="button">
          ×
        </button>
      )}
    </div>
  );
};

export default Alert;
