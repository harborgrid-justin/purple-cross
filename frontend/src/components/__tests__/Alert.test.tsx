/**
 * WF-COMP-XXX | Alert.test.tsx - Alert Component Tests
 * Purpose: Comprehensive tests for Alert component
 * Dependencies: @testing-library/react, @testing-library/user-event
 * Last Updated: 2025-10-24 | File Type: .tsx
 */

import { describe, it, expect, vi } from 'vitest';
import { renderWithoutRouter, createUser } from '@/test-utils';
import Alert from '../Alert';

describe('Alert Component', () => {
  describe('Rendering', () => {
    it('should render with default props', () => {
      const { getByRole } = renderWithoutRouter(<Alert>Alert message</Alert>);

      const alert = getByRole('alert');
      expect(alert).toBeInTheDocument();
      expect(alert).toHaveTextContent('Alert message');
    });

    it('should render children correctly', () => {
      const { getByText } = renderWithoutRouter(
        <Alert>This is an important message</Alert>
      );

      expect(getByText('This is an important message')).toBeInTheDocument();
    });

    it('should have correct default classes', () => {
      const { getByRole } = renderWithoutRouter(<Alert>Message</Alert>);

      const alert = getByRole('alert');
      expect(alert).toHaveClass('alert');
      expect(alert).toHaveClass('alert-info'); // Default type
    });

    it('should render with custom className', () => {
      const { getByRole } = renderWithoutRouter(
        <Alert className="custom-alert">Message</Alert>
      );

      const alert = getByRole('alert');
      expect(alert).toHaveClass('custom-alert');
      expect(alert).toHaveClass('alert'); // Should still have base class
    });
  });

  describe('Alert Types', () => {
    it('should render success alert', () => {
      const { getByRole } = renderWithoutRouter(
        <Alert type="success">Success message</Alert>
      );

      expect(getByRole('alert')).toHaveClass('alert-success');
    });

    it('should render error alert', () => {
      const { getByRole } = renderWithoutRouter(
        <Alert type="error">Error message</Alert>
      );

      expect(getByRole('alert')).toHaveClass('alert-error');
    });

    it('should render warning alert', () => {
      const { getByRole } = renderWithoutRouter(
        <Alert type="warning">Warning message</Alert>
      );

      expect(getByRole('alert')).toHaveClass('alert-warning');
    });

    it('should render info alert (default)', () => {
      const { getByRole } = renderWithoutRouter(<Alert>Info message</Alert>);

      expect(getByRole('alert')).toHaveClass('alert-info');
    });

    it('should render info alert explicitly', () => {
      const { getByRole } = renderWithoutRouter(
        <Alert type="info">Info message</Alert>
      );

      expect(getByRole('alert')).toHaveClass('alert-info');
    });
  });

  describe('Icons', () => {
    it('should render success icon', () => {
      const { container } = renderWithoutRouter(
        <Alert type="success">Success</Alert>
      );

      const icon = container.querySelector('.alert-icon');
      expect(icon).toBeInTheDocument();
      expect(icon).toHaveTextContent('✓');
    });

    it('should render error icon', () => {
      const { container } = renderWithoutRouter(<Alert type="error">Error</Alert>);

      const icon = container.querySelector('.alert-icon');
      expect(icon).toBeInTheDocument();
      expect(icon).toHaveTextContent('✕');
    });

    it('should render warning icon', () => {
      const { container } = renderWithoutRouter(
        <Alert type="warning">Warning</Alert>
      );

      const icon = container.querySelector('.alert-icon');
      expect(icon).toBeInTheDocument();
      expect(icon).toHaveTextContent('⚠');
    });

    it('should render info icon', () => {
      const { container } = renderWithoutRouter(<Alert type="info">Info</Alert>);

      const icon = container.querySelector('.alert-icon');
      expect(icon).toBeInTheDocument();
      expect(icon).toHaveTextContent('ℹ');
    });

    it('should have aria-hidden on icon', () => {
      const { container } = renderWithoutRouter(<Alert>Message</Alert>);

      const icon = container.querySelector('.alert-icon');
      expect(icon).toHaveAttribute('aria-hidden', 'true');
    });
  });

  describe('Alert Content', () => {
    it('should render alert content div', () => {
      const { container } = renderWithoutRouter(<Alert>Content</Alert>);

      const content = container.querySelector('.alert-content');
      expect(content).toBeInTheDocument();
      expect(content).toHaveTextContent('Content');
    });

    it('should render simple text content', () => {
      const { getByText } = renderWithoutRouter(
        <Alert>Simple text message</Alert>
      );

      expect(getByText('Simple text message')).toBeInTheDocument();
    });

    it('should render complex JSX content', () => {
      const { getByText } = renderWithoutRouter(
        <Alert>
          <strong>Important:</strong> This is a complex message with{' '}
          <a href="/link">a link</a>
        </Alert>
      );

      expect(getByText('Important:')).toBeInTheDocument();
      expect(getByText('a link')).toBeInTheDocument();
    });

    it('should render multiline content', () => {
      const { getByText } = renderWithoutRouter(
        <Alert>
          <div>Line 1</div>
          <div>Line 2</div>
          <div>Line 3</div>
        </Alert>
      );

      expect(getByText('Line 1')).toBeInTheDocument();
      expect(getByText('Line 2')).toBeInTheDocument();
      expect(getByText('Line 3')).toBeInTheDocument();
    });
  });

  describe('Close Button', () => {
    it('should render close button when onClose is provided', () => {
      const handleClose = vi.fn();
      const { getByRole } = renderWithoutRouter(
        <Alert onClose={handleClose}>Message</Alert>
      );

      expect(getByRole('button', { name: /close alert/i })).toBeInTheDocument();
    });

    it('should not render close button when onClose is not provided', () => {
      const { queryByRole } = renderWithoutRouter(<Alert>Message</Alert>);

      expect(queryByRole('button', { name: /close alert/i })).not.toBeInTheDocument();
    });

    it('should have correct aria-label on close button', () => {
      const handleClose = vi.fn();
      const { getByRole } = renderWithoutRouter(
        <Alert onClose={handleClose}>Message</Alert>
      );

      const closeButton = getByRole('button', { name: /close alert/i });
      expect(closeButton).toHaveAttribute('aria-label', 'Close alert');
    });

    it('should have correct type on close button', () => {
      const handleClose = vi.fn();
      const { getByRole } = renderWithoutRouter(
        <Alert onClose={handleClose}>Message</Alert>
      );

      const closeButton = getByRole('button', { name: /close alert/i });
      expect(closeButton).toHaveAttribute('type', 'button');
    });

    it('should have correct class on close button', () => {
      const handleClose = vi.fn();
      const { getByRole } = renderWithoutRouter(
        <Alert onClose={handleClose}>Message</Alert>
      );

      const closeButton = getByRole('button', { name: /close alert/i });
      expect(closeButton).toHaveClass('alert-close');
    });

    it('should display × symbol in close button', () => {
      const handleClose = vi.fn();
      const { getByRole } = renderWithoutRouter(
        <Alert onClose={handleClose}>Message</Alert>
      );

      const closeButton = getByRole('button', { name: /close alert/i });
      expect(closeButton).toHaveTextContent('×');
    });

    it('should call onClose when close button is clicked', async () => {
      const handleClose = vi.fn();
      const user = createUser();
      const { getByRole } = renderWithoutRouter(
        <Alert onClose={handleClose}>Message</Alert>
      );

      const closeButton = getByRole('button', { name: /close alert/i });
      await user.click(closeButton);

      expect(handleClose).toHaveBeenCalledTimes(1);
    });

    it('should call onClose multiple times on multiple clicks', async () => {
      const handleClose = vi.fn();
      const user = createUser();
      const { getByRole } = renderWithoutRouter(
        <Alert onClose={handleClose}>Message</Alert>
      );

      const closeButton = getByRole('button', { name: /close alert/i });
      await user.click(closeButton);
      await user.click(closeButton);
      await user.click(closeButton);

      expect(handleClose).toHaveBeenCalledTimes(3);
    });

    it('should support keyboard interaction on close button (Enter)', async () => {
      const handleClose = vi.fn();
      const user = createUser();
      const { getByRole } = renderWithoutRouter(
        <Alert onClose={handleClose}>Message</Alert>
      );

      const closeButton = getByRole('button', { name: /close alert/i });
      closeButton.focus();
      await user.keyboard('{Enter}');

      expect(handleClose).toHaveBeenCalledTimes(1);
    });

    it('should support keyboard interaction on close button (Space)', async () => {
      const handleClose = vi.fn();
      const user = createUser();
      const { getByRole } = renderWithoutRouter(
        <Alert onClose={handleClose}>Message</Alert>
      );

      const closeButton = getByRole('button', { name: /close alert/i });
      closeButton.focus();
      await user.keyboard(' ');

      expect(handleClose).toHaveBeenCalledTimes(1);
    });
  });

  describe('Layout and Structure', () => {
    it('should have proper DOM structure', () => {
      const { container } = renderWithoutRouter(<Alert>Message</Alert>);

      const alert = container.querySelector('.alert');
      const icon = container.querySelector('.alert-icon');
      const content = container.querySelector('.alert-content');

      expect(alert).toContainElement(icon);
      expect(alert).toContainElement(content as HTMLElement);
    });

    it('should have proper DOM structure with close button', () => {
      const handleClose = vi.fn();
      const { container } = renderWithoutRouter(
        <Alert onClose={handleClose}>Message</Alert>
      );

      const alert = container.querySelector('.alert');
      const icon = container.querySelector('.alert-icon');
      const content = container.querySelector('.alert-content');
      const closeButton = container.querySelector('.alert-close');

      expect(alert).toContainElement(icon);
      expect(alert).toContainElement(content as HTMLElement);
      expect(alert).toContainElement(closeButton);
    });

    it('should order elements correctly: icon, content, close button', () => {
      const handleClose = vi.fn();
      const { container } = renderWithoutRouter(
        <Alert onClose={handleClose}>Message</Alert>
      );

      const alert = container.querySelector('.alert');
      const children = Array.from(alert?.children || []);

      expect(children[0]).toHaveClass('alert-icon');
      expect(children[1]).toHaveClass('alert-content');
      expect(children[2]).toHaveClass('alert-close');
    });
  });

  describe('Accessibility', () => {
    it('should have role="alert"', () => {
      const { getByRole } = renderWithoutRouter(<Alert>Message</Alert>);

      expect(getByRole('alert')).toBeInTheDocument();
    });

    it('should be automatically announced by screen readers', () => {
      const { getByRole } = renderWithoutRouter(
        <Alert>Important notification</Alert>
      );

      const alert = getByRole('alert');
      expect(alert).toBeInTheDocument();
    });

    it('should have accessible close button', () => {
      const handleClose = vi.fn();
      const { getByRole } = renderWithoutRouter(
        <Alert onClose={handleClose}>Message</Alert>
      );

      const closeButton = getByRole('button', { name: /close alert/i });
      expect(closeButton).toHaveAccessibleName('Close alert');
    });

    it('should hide icon from screen readers', () => {
      const { container } = renderWithoutRouter(<Alert>Message</Alert>);

      const icon = container.querySelector('.alert-icon');
      expect(icon).toHaveAttribute('aria-hidden', 'true');
    });

    it('should be keyboard navigable when closeable', async () => {
      const handleClose = vi.fn();
      const user = createUser();
      const { getByRole } = renderWithoutRouter(
        <Alert onClose={handleClose}>Message</Alert>
      );

      const closeButton = getByRole('button', { name: /close alert/i });

      // Tab to focus the button
      await user.tab();
      expect(closeButton).toHaveFocus();
    });
  });

  describe('Different Alert Scenarios', () => {
    it('should render success alert with close button', () => {
      const handleClose = vi.fn();
      const { getByRole, container } = renderWithoutRouter(
        <Alert type="success" onClose={handleClose}>
          Operation completed successfully
        </Alert>
      );

      expect(getByRole('alert')).toHaveClass('alert-success');
      expect(container.querySelector('.alert-icon')).toHaveTextContent('✓');
      expect(getByRole('button', { name: /close alert/i })).toBeInTheDocument();
    });

    it('should render error alert with custom class', () => {
      const { getByRole } = renderWithoutRouter(
        <Alert type="error" className="critical-error">
          Critical error occurred
        </Alert>
      );

      const alert = getByRole('alert');
      expect(alert).toHaveClass('alert-error');
      expect(alert).toHaveClass('critical-error');
    });

    it('should render warning alert with complex content', () => {
      const { getByText } = renderWithoutRouter(
        <Alert type="warning">
          <div>
            <strong>Warning:</strong> This action cannot be undone.
          </div>
          <div>Are you sure you want to continue?</div>
        </Alert>
      );

      expect(getByText('Warning:')).toBeInTheDocument();
      expect(getByText('This action cannot be undone.')).toBeInTheDocument();
    });
  });

  describe('Edge Cases', () => {
    it('should handle empty children', () => {
      const { getByRole } = renderWithoutRouter(<Alert />);

      expect(getByRole('alert')).toBeInTheDocument();
    });

    it('should handle very long messages', () => {
      const longMessage =
        'This is a very long alert message that might span multiple lines and needs to be handled gracefully by the component without breaking the layout or causing any visual issues';
      const { getByText } = renderWithoutRouter(<Alert>{longMessage}</Alert>);

      expect(getByText(longMessage)).toBeInTheDocument();
    });

    it('should handle messages with special characters', () => {
      const specialMessage = 'Alert with special chars: <>&"\'';
      const { getByText } = renderWithoutRouter(<Alert>{specialMessage}</Alert>);

      expect(getByText(specialMessage)).toBeInTheDocument();
    });

    it('should handle null onClose gracefully', () => {
      const { queryByRole } = renderWithoutRouter(
        <Alert onClose={undefined}>Message</Alert>
      );

      expect(queryByRole('button')).not.toBeInTheDocument();
    });
  });

  describe('Snapshot Tests', () => {
    it('should match snapshot for default alert', () => {
      const { container } = renderWithoutRouter(<Alert>Default alert</Alert>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for success alert with close button', () => {
      const handleClose = vi.fn();
      const { container } = renderWithoutRouter(
        <Alert type="success" onClose={handleClose}>
          Success alert
        </Alert>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for error alert', () => {
      const { container } = renderWithoutRouter(
        <Alert type="error">Error alert</Alert>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for warning alert', () => {
      const { container } = renderWithoutRouter(
        <Alert type="warning">Warning alert</Alert>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
