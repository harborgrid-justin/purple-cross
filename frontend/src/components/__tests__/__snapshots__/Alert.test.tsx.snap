// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert Component > Snapshot Tests > should match snapshot for default alert 1`] = `
<div
  class="alert alert-info"
  role="alert"
>
  <span
    aria-hidden="true"
    class="alert-icon"
  >
    ℹ
  </span>
  <div
    class="alert-content"
  >
    Default alert
  </div>
</div>
`;

exports[`Alert Component > Snapshot Tests > should match snapshot for error alert 1`] = `
<div
  class="alert alert-error"
  role="alert"
>
  <span
    aria-hidden="true"
    class="alert-icon"
  >
    ✕
  </span>
  <div
    class="alert-content"
  >
    Error alert
  </div>
</div>
`;

exports[`Alert Component > Snapshot Tests > should match snapshot for success alert with close button 1`] = `
<div
  class="alert alert-success"
  role="alert"
>
  <span
    aria-hidden="true"
    class="alert-icon"
  >
    ✓
  </span>
  <div
    class="alert-content"
  >
    Success alert
  </div>
  <button
    aria-label="Close alert"
    class="alert-close"
    type="button"
  >
    ×
  </button>
</div>
`;

exports[`Alert Component > Snapshot Tests > should match snapshot for warning alert 1`] = `
<div
  class="alert alert-warning"
  role="alert"
>
  <span
    aria-hidden="true"
    class="alert-icon"
  >
    ⚠
  </span>
  <div
    class="alert-content"
  >
    Warning alert
  </div>
</div>
`;
